### PR TITLE
Implement some FRs

### PR DESCRIPTION
**Pull request by sandwwraith**
* Android NDK targets
* Rename UnitSerializer to Unit.serializer

### CODE REVIEW

File: runtime/commonMain/src/kotlinx/serialization/builtins/PrimitiveSerializers.kt [hunk 1]
```diff
@@ -99,10 +99,20 @@ public fun Boolean.Companion.serializer(): KSerializer<Boolean> = kotlinx.serial
...
 /**
- * Returns serializer for [UnitSerializer] with [descriptor][SerialDescriptor] of [StructureKind.OBJECT] kind.
+ * Returns serializer for [Unit] with [descriptor][SerialDescriptor] of [StructureKind.OBJECT] kind.
  */
+@Deprecated(
+    message = "Deprecated in the favour of Unit.serializer() extension",
+    level = DeprecationLevel.ERROR,
+    replaceWith = ReplaceWith("Unit.serializer()", imports = ["kotlinx.serialization.builtins.serializer"])
+)
 public fun UnitSerializer(): KSerializer<Unit> = kotlinx.serialization.internal.UnitSerializer
```
Comment (qwwdfsad) [inline runtime/commonMain/src/kotlinx/serialization/builtins/PrimitiveSerializers.kt:109]: Please move this declaration to Migration.kt to keep them in one pleace

File: gradle/native_mpp.gradle [hunk 2]
```diff
@@ -45,6 +45,12 @@ kotlin {
 
             // WASM
             addTarget(presets.wasm32)
+
+            // Android NDK
+            addTarget(presets.androidNativeArm32)
+            addTarget(presets.androidNativeArm64)
+            addTarget(presets.androidNativeX86)
+            addTarget(presets.androidNativeX64)
```
Comment (qwwdfsad) [inline gradle/native_mpp.gradle:53]: Have you checked that all these targets are being tested on our TeamCity, can be depend upon and are generally valid?

Comment (sandwwraith) [reply gradle/native_mpp.gradle:53]: It builds locally, and it builds on TeamCity. Not sure if we have any dedicated Gradle tools to test any targets except the host and iOS ones.

**Comment by sandwwraith** [comment]
@senagbe You can checkout branch from this PR and build it with `./gradlew publishToMavenLocal -Pnative.deploy=true`. Android targets should be available in maven local after that.

File: runtime/commonMain/src/kotlinx/serialization/builtins/PrimitiveSerializers.kt (hunk 1 above)
Comment (sandwwraith) [reply runtime/commonMain/src/kotlinx/serialization/builtins/PrimitiveSerializers.kt:109]: There's no Migration.kt in builtins package

**Comment by qwwdfsad** [comment]
Please merge with a sqaush

**Comment by senagbe** [comment]
@sandwwraith I see #862 this has been reverted. Did it fail tests? How can I help?
Also, should we re-open the issue?

**Comment by sandwwraith** [comment]
@senagbe Unfortunately, there is no testing infrastructure for android native targets at this moment. Since our goal is to provide a quality library, publishing artifacts that have not been tested is a bad practice. You still can build and use them if they work for you, without particular guarantees.

**Comment by senagbe** [comment]
Thanks @sandwwraith I'm going to look at how to add tests for this when I have some bandwidth. I'd expect them to be essentially identical to the Linux tests.

**Pull request merged**